frontend/app/controllers/spree/taxons_controller.rb

```diff
@@ -4,19 +4,22 @@ module Spree
   class TaxonsController < Spree::StoreController
     helper 'spree/products', 'spree/taxon_filters'
 
+    before_action :load_taxon, only: [:show]
+
     respond_to :html
 
     def show
-      @taxon = Spree::Taxon.find_by!(permalink: params[:id])
-      return unless @taxon
```

Comment:
This line was useless, right?

Reply:
yes. exactly. I also double checked to be sure but the `find_by!` raises an exception if the element is not found, returning 404.